static/app/components/forms/compactSelect.tsx

```diff
@@ -125,6 +130,82 @@ export const CompactSelectControl = ({
   );
 };
 
+// TODO(vl): Turn this into a reusable component
+function Menu({
+  // Trigger & trigger state
+  targetRef,
+  onClose,
+  // Overlay props
+  offset = 8,
+  crossOffset = 0,
+  containerPadding = 8,
+  placement = 'bottom left',
+  shouldCloseOnBlur = true,
+  isDismissable = true,
+  maxMenuHeight = 400,
+  children,
+}: MenuProps) {
+  // Control the overlay's position
+  const overlayRef = useRef<HTMLDivElement>(null);
+  const {overlayProps} = useOverlay(
+    {
+      isOpen: true,
+      onClose,
+      shouldCloseOnBlur,
+      isDismissable,
+      shouldCloseOnInteractOutside: target =>
+        target && targetRef.current !== target && !targetRef.current?.contains(target),
+    },
+    overlayRef
+  );
+  const {overlayProps: positionProps} = useOverlayPosition({
```

Comment:
Is this our first usage of `useOverlayPosition` 😬

Not sure if I'll end up landing this for things like our tooltips, popper still seems to have a bit of an edge unfortuantely.

Reply:
Yepp, and this is actually our second use. The first one was in `DropdownMenuV2`. If we end up using PopperJS or sth else I'll quickly convert these two over.